lib/dialects/abstract/query-generator.js

```diff
@@ -1386,16 +1386,34 @@ const QueryGenerator = {
         }
         mainQueryItems.push(this.selectFromTableFragment(options, model, mainAttributes, '('+
           options.groupedLimit.values.map(value => {
-            const where = _.assign({}, options.where);
-            where[options.groupedLimit.on] = value;
+            let where = options.where || {}
+              , include;
+
+            if (Array.isArray(options.groupedLimit.on)) {
+              include = Model._validateIncludedElements({
+                include: [{
+                  attributes: [],
+                  association: options.groupedLimit.on[0],
+                  where: {
+                    [options.groupedLimit.on[1]]: value
+                  }
+                }],
+                model
+              }).include;
+            } else {
+              where = _.assign(options.where);
+              where[options.groupedLimit.on] = value;
+            }
```

Comment:
We can't do an include on the main query (since a main include should be added to the main query :P)

We could add `groupedLimit.include`, but then we would have to do `options.groupedLimit.include[0].where[options.groupedLimit.on] = value` in the loop. Not sure I think thats prettier.